classes/logging_request_info.class.php

```diff
@@ -0,0 +1,125 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Submission class for plagiarism_turnitinsim component.
+ *
+ * @package   plagiarism_turnitinsim
+ * @copyright 2017 Turnitin
+ * @author    John McGettrick <jmcgettrick@turnitin.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+require_once($CFG->dirroot . '/plagiarism/turnitinsim/lib.php');
+
+
+/**
+ * Class models request info sent to be logged.
+ */
+class logging_request_info {
+
+    /**
+     * @var string The url.
+     */
+    private $url;
+
+    /**
+     * @var string The Http Method.
+     */
+    private $method;
+
+    /**
+     * @var array Request headers.
+     */
+    private $headers;
+
+    /**
+     * @var int Response http status.
+     */
+    private $response_status;
```

Comment:
These variables will need to be without underscores. Did you run the Moodle code checker? It's different to the PHP Doc checker.

Reply:
Yeah, I have checked the Moodle code checker and didn't get any warning.
I have not run Php Doc checker, I will do that and will resolve all the warnings.

Reply:
That's surprising. I know when we used Travis code checker it used to flag snake case variables as problems. I guess the code checker doesn't detect all the things Moodle don't like. If you check https://docs.moodle.org/dev/Coding_style#Variables Moodle say anything other than lowercase variable names are bad. I disagree with it as it makes it less readable but we usually go with what Moodle say.

Reply:
Resolved